tests/integration/check-your-answers/file-upload-site-details.test.js

```diff
@@ -0,0 +1,154 @@
+      validatePublicRegister(document, expectedPageContent)
+      validateSubmissionSection(document, expectedPageContent)
+    }
+  )
+
+  const getPageDocument = async (exemption) => {
+    jest.spyOn(cacheUtils, 'getExemptionCache').mockReturnValue(exemption)
+    jest.spyOn(authRequests, 'authenticatedGetRequest').mockResolvedValue({
+      payload: { value: { taskList: { id: exemption.id } } }
+    })
+
+    const response = await server.inject({
+      method: 'GET',
+      url: routes.CHECK_YOUR_ANSWERS
+    })
+
+    expect(response.statusCode).toBe(statusCodes.ok)
+    return new JSDOM(response.result).window.document
+  }
+
+  const validatePageStructure = (document, expected) => {
+    const heading = document.querySelector('#check-your-answers-heading')
+    expect(heading.textContent.trim()).toBe(expected.pageTitle)
+
+    const backLink = document.querySelector('.govuk-back-link')
+    expect(backLink.textContent.trim()).toBe(expected.backLinkText)
+  }
+
+  const validateAllSummaryCardsExist = (document, expected) => {
+    expected.summaryCards.forEach((expectedTitle) => {
+      const cardTitles = document.querySelectorAll('.govuk-summary-card__title')
+      const foundCard = Array.from(cardTitles).find(
+        (title) => title.textContent.trim() === expectedTitle
+      )
+      expect(foundCard).toBeTruthy()
+    })
+  }
+
+  const validateSummaryCardContent = (
+    document,
+    cardSelector,
+    expectedContent
+  ) => {
+    const card = document.querySelector(cardSelector)
+    Object.entries(expectedContent).forEach(([key, value]) => {
+      const rows = card.querySelectorAll('.govuk-summary-list__row')
+      const row = Array.from(rows).find((row) => {
+        const keyElement = row.querySelector('.govuk-summary-list__key')
+        return keyElement && keyElement.textContent.trim() === key
+      })
+      expect(row).toBeTruthy()
+      const valueElement = row.querySelector('.govuk-summary-list__value')
+      expect(valueElement.textContent.trim()).toBe(value)
+    })
+  }
+
+  const validateProjectDetails = (document, expected) => {
+    validateSummaryCardContent(
+      document,
+      '#project-details-card',
+      expected.projectDetails
+    )
+  }
+
+  const validateActivityDates = (document, expected) => {
+    validateSummaryCardContent(
+      document,
+      '#activity-dates-card',
+      expected.activityDates
+    )
+  }
+
+  const validateActivityDetails = (document, expected) => {
+    validateSummaryCardContent(
+      document,
+      '#activity-details-card',
+      expected.activityDetails
+    )
+  }
+
+  const validateSiteDetails = (document, expected) => {
+    validateSummaryCardContent(
+      document,
+      '#site-details-card',
+      expected.siteDetails
+    )
+
+    if (expected.shouldNotHaveMapView) {
+      const siteCard = document.querySelector('#site-details-card')
+      const rows = siteCard.querySelectorAll('.govuk-summary-list__row')
```

Comment:
This could be extracted to a `getSummaryListRowValue` helper outside this file which takes the key label as a param?